### PR TITLE
Extended RenderGlyphs method

### PR DESCRIPTION
**Pull request by intercross21**
Now it accepts the ExtendedSet data container which allows to modify set of characters which will be drawn.
http://forum.adamslair.net/viewtopic.php?f=4&t=715#p4282

### CODE REVIEW

File: DualityEditor/Utility/Extensions/ExtMethodsDualityFont.cs [hunk 1]
```diff
@@ -113,12 +114,25 @@ public static void RenderGlyphs(this DualityFont font, FontFamily fontFamily)
...
-		private static void RenderGlyphs(DualityFont target, SysDrawFont internalFont)
+		private static void RenderGlyphs(DualityFont target, SysDrawFont internalFont, ExtendedCharSet extendedSet = null)
 		{
-			DualityFont.GlyphData[] glyphs = new DualityFont.GlyphData[DefaultChars.Length];
+            string allChars = DefaultChars;
+            string allCharBaseLineRef = CharBaseLineRef;
+            string allCharDescentRef = CharDescentRef;
+            string allCharBodyAscentRef = CharBodyAscentRef;
+
+            if(extendedSet != null)
```
Comment (ilexp) [inline DualityEditor/Utility/Extensions/ExtMethodsDualityFont.cs]: Code Style: Forgot to add a space between `if` and `(`.

File: DualityEditor/Utility/Extensions/ExtMethodsDualityFont.cs [hunk 2]
```diff
@@ -113,12 +114,25 @@ public static void RenderGlyphs(this DualityFont font, FontFamily fontFamily)
...
+            {
+                allChars += extendedSet.Chars;                      //Maybe we should check if any char is duplicated
+                allCharBaseLineRef += extendedSet.CharBaseLine;     //And also check availbility of those characters in current font
```
Comment (ilexp) [inline DualityEditor/Utility/Extensions/ExtMethodsDualityFont.cs]: Code Style: Avoid comments at the end of a code line. Also add a space after `//` before the text begins.

**Comment by ilexp** [comment]
Good addition! Some review in addition to inline comments:

- Please use tabs for indentation, rather than spaces.
- Replace public fields in `ExtendedCharSet` with public properties.
- Please provide XML documentation for every public property of `ExtendedCharSet`, as well as the class as a whole. The editor is chronically under-documented and I'm trying to improve here.
- Rename `ExtendedCharSet` to something like `FontRenderGlyphCharSet` to reflect its degree of specialization and provide context. It's in a very open namespace.

**Comment by intercross21** [comment]
I changed few options in VS13 so now auto-correction should work properly.

**Comment by ilexp** [comment]
Thanks! Lines 61, 110 and 120-123 still use spaces instead of tabs - I'll fix this when I do the merge, but please double-check next time :)

**Pull request merged**